src/java/org/apache/cassandra/index/internal/composites/CompositesSearcher.java

```diff
@@ -231,16 +231,27 @@ private IndexEntry findEntry(Clustering clustering)
                 while (entriesIdx < entries.size())
                 {
                     IndexEntry entry = entries.get(entriesIdx++);
+                    Clustering indexedEntryClustering = entry.indexedEntryClustering;
                     // The entries are in clustering order. So that the requested entry should be the
                     // next entry, the one at 'entriesIdx'. However, we can have stale entries, entries
                     // that have no corresponding row in the base table typically because of a range
                     // tombstone or partition level deletion. Delete such stale entries.
-                    int cmp = comparator.compare(entry.indexedEntryClustering, clustering);
+                    int cmp = comparator.compare(indexedEntryClustering, clustering);
                     assert cmp <= 0; // this would means entries are not in clustering order, which shouldn't happen
                     if (cmp == 0)
                         return entry;
                     else
-                        staleEntries.add(entry);
+                    {
+                        boolean isStaticClustering = dataIter.metadata().hasStaticColumns();
```

Comment:
Might read a bit awkward, but given I am in a hot path I wanted to incur the cost to test for nulls only _if_ needed

Reply:
It is effectively a bit hard to read. I would have done something like:
boolean hasStaticColumns = dataIter.metadata().hasStaticColumns();
```

			// COMPACT COMPOSITE tables support null values in there clustering key but
                        // those tables do not support static columns. By consequence if a table
                        // has some static columns and all its clustering key elements are null
                        // it means that the partition exists and contains only static data 
                        if (!hasStaticColumns() || !containsOnlyNullValues(indexedEntryClustering) 
                             staleEntries.add(entry);
```
What do you think?

Reply:
@blerer if you give me thumbs up I'll rebase, squash, forward merge and run CI before handing it over to you for committing.